independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -1026,22 +1039,30 @@ private List<BeanInfo> findBeans(Collection<DotName> beanDefiningAnnotations, Li
                         syncObserverMethods.computeIfAbsent(method, ignored -> new HashSet<>())
                                 .add(beanClass);
                         if (!Modifier.isAbstract(beanClass.flags())) {
-                            // add only concrete classes
```

Comment:
Nitpicking: this comment clearly refers to the previous line (checking if the class is abstract or "concrete"). The line is moved pretty far away and it's no longer exactly clear what "add only concrete classes" means.